### PR TITLE
Add Application load balancer resource, tests, docs and tf

### PR DESCRIPTION
**Pull request by dmurray-chef**
Signed-off-by: Darren Murray <dmurray@chef.io>

### Description

Add new resource ALB. Integration and unit tests. Add docs and terraform for alb creation

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/integration/build/aws.tf [hunk 1]
```diff
@@ -1304,4 +1322,28 @@ resource "aws_dynamodb_table" "aws-dynamodb-table" {
...
+  vpc_id      = "${aws_vpc.eks_vpc.id}"
+
+  tags = {
+    Name        = "${var.aws_security_group_lb}"
+    Environment = "Dev"
+  }
+}
+
+resource "aws_lb" "aws-alb" {
+  name               = "${var.aws_alb_name}"
```
Comment (skpaterson) [inline test/integration/build/aws.tf]: Maybe missing a `  count       = "${var.aws_enable_creation}"` here?

File: libraries/aws_alb.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      @vpc_id                   = alb.vpc_id
+    end
+  end
+
+  def exists?
+    !@load_balancer_arn.nil?
+  end
+
+  def to_s
+    "AWS ALB #{load_balancer_name}"
```
Comment (james-stocks) [inline libraries/aws_alb.rb]: Should this be `@load_balancer_name` ?

File: libraries/aws_alb.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    describe aws_alb('arn:aws:elasticloadbalancing') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :availability_zones, :canonical_hosted_zone_id, :created_time, :dns_name, :load_balancer_arn,
+              :load_balancer_name, :load_balancer_addresses, :scheme, :security_groups, :state, :subnets, :type, :vpc_id, :zone_names
+
+  def initialize(opts = {})
+    opts = { load_balancer_arns: opts } if opts.is_a?(String)
```
Comment (james-stocks) [inline libraries/aws_alb.rb]: This creates the key `load_balancer_arns` but the rest of the class seems to work with the opts key `load_balancer_arn`
(if I'm just misreading this, sorry)

Comment (skpaterson) [reply libraries/aws_alb.rb]: that's a good catch @james-stocks - we should have a failing unit test for the `String` case ...

Comment (dmurray-chef) [reply libraries/aws_alb.rb]: Good spot! I've added a unit test for this too.

File: libraries/aws_alb.rb (hunk 2 above)
Comment (dmurray-chef) [reply libraries/aws_alb.rb]: fixed

File: test/integration/build/aws.tf (hunk 1 above)
Comment (dmurray-chef) [reply test/integration/build/aws.tf]: fixed

**Pull request merged**